### DISCUSSION TITLE
Tower Defense - UCRS

### DISCUSSION THREAD

**Discussion by TimWaeckerle** [Freaky2164/TowerDefense#6] [General]
Hi all!

TD Updates!

We updated some of our Use Cases!

- [Create Game](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Use-Cases/Create%20Game.md) 
- [Start Round](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Use-Cases/Start%20Round.md)
- [Place Towers](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Use-Cases/Place%20Towers.md)

Also check out our new UCRS!

- [Start Round](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Use-Cases/Realization%20Specification/UCRS%20%232.md)
- [Create Game & Load Game](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Use-Cases/Realization%20Specification/UCRS%20%231.md)

Thanks for your contribution!

Tower Defense Team

**Comment by TeamTuDu** [comment]
Hi TowerDefense Team!

we think your project is quite interesting and the idea of just loading up a previously started game is a really cool touch!

Some points regarding your use cases that we stumbled upon:

regarding the start round use case:
- check your arrow labels, "isAutostart" has 2x "no" as options ;) 
- by swapping the order of "isAutoStart" and "round is finished" you could simplify your diagram and make it a bit more intuitive :)

	
regarding "place towers"

- as we currently understand it, you want to select a tower, then move the tower to an appropiate position and finally confirm the placing with "place tower" (Y/N-Style confirmation) 
- maybe you could define "wait for user action" a bit clearer as we were not sure what user action is being waited for. You could also connect it to the next-in-line user interactions, this would improve clarity.
- we were a bit confused what you want to check with "is placed"; if "is placed" is your (Y/N) decision, then we don't really understand why it has a connection from "object tower is created".

But besides that we really liked the simple, high level approach you chose for you charts!

Two questions that arose while thinking about your designs a bit:
- Do your towers have any aim funtionality or do they fire constantly at preset directions?
- Do you want to implement only hitscan towers like lasers or do you want to use towers with travel time as well?

Looking foward to seeing more of your project, a really cool idea!

Best regards
TuDu

**Comment by dantschgl** [comment]
Hi Tower Defense Team,

we like your project idea and think you've done a good job so far.

In your UCRSs we found boxes labeled "Alt" and couldn't figure out what you meant by that. Could you help us out with that?

In one of your Flow of events (Use Case Start Round) we found a small problem: you mislabeled two if-queries (attached picture)

![grafik](https://user-images.githubusercontent.com/115186421/199745475-f075c93d-5507-4acd-8978-63a74b4a6d5c.png)

Other than that we think you really did a great job and everything is easily understandable. Keep up the good work 👍 

~GuessMaster
